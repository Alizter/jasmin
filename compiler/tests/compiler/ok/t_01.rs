param n : u64;

// we can use a variable name that is ignored
fn foo1(x : stack u64) -> stack u64 * reg u64 * reg bool;

// we can also leave out variable names
fn foo2(stack u64, stack u64[n], x,x,x : stack u64) -> stack u64[n] =
  python print_foo;

// nothing
fn foo3(_x : stack u64) {
}

// decl only
fn foo4(_x : stack u64) {
  _y : stack u64; // will not be printed
}

// body only
fn foo5(x : stack u64) {
  x += x;
}

// return only
fn foo6(x : stack u64) -> stack u64 {
  return x;
}

// deck + body
fn foo7(x : stack u64) {
  y : stack u64;
  x += y;
}

// deck + return
fn foo8(x : stack u64) -> stack u64 * stack u64 {
  y : stack u64;
  return y, x;
}

// body + return
fn foo9(x : stack u64) -> stack u64 {
  x += x;
  return x;
}

// decl + body + return
fn foo10(x, y : stack u64, z : reg bool) -> stack u64 {
  w : stack u64;
  w = x;
  w += x;
  if (w = 5) {
    z, x += w;
    z, x += y;
  }
  return x;
}

/*
START:CMD
ARG="print[input][types]"
END:CMD
*/
