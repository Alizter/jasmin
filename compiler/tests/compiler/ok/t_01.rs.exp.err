   Compiling test_jasmin v0.1.0 (file:///Users/beschmi/git/jasmin/compiler/tests/build)
    Finished debug [unoptimized + debuginfo] target(s) in REMOVED_TIME
     Running target/debug/deps/test_jasmin-2d9df3cd2a86189e
   Doc-tests test_jasmin
Processed file tests/compiler/ok/t_01.rs
