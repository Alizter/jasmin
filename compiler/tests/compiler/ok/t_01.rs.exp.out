>> input:
#[macro_use] extern crate jasmin;
param n : b64;

fn foo1(stack! (b64)) -> (stack! (b64),reg! (b64),reg! (b1));

fn foo2(stack! (b64),stack! (b64[(n: b64)]),stack! (b64),stack! (b64),stack! (b64)) -> stack! (b64[(n: b64)]) = python print_foo;

fn foo3(mut _x: stack! (b64)) {
}

fn foo4(mut _x: stack! (b64)) {
}

fn foo5(mut x: stack! (b64)) {
    code! {
        x: stack! (b64) += x: stack! (b64);
    }
}

fn foo6(mut x: stack! (b64)) -> stack! (b64) {
    return x: stack! (b64)
}

fn foo7(mut x: stack! (b64)) {
    var! {
        y: stack! (b64);
    }
    code! {
        x: stack! (b64) += y: stack! (b64);
    }
}

fn foo8(mut x: stack! (b64)) -> (stack! (b64),stack! (b64)) {
    var! {
        y: stack! (b64);
    }
    return (y: stack! (b64),x: stack! (b64))
}

fn foo9(mut x: stack! (b64)) -> stack! (b64) {
    code! {
        x: stack! (b64) += x: stack! (b64);
    }
    return x: stack! (b64)
}

fn foo10(mut x: stack! (b64), mut y: stack! (b64), mut z: reg! (b1)) -> stack! (b64) {
    var! {
        w: stack! (b64);
    }
    code! {
        w: stack! (b64) = x: stack! (b64);
        w: stack! (b64) += x: stack! (b64);
        if (w: stack! (b64) == jc!(5)) {
            z: reg! (b1), x: stack! (b64) += w: stack! (b64);
            z: reg! (b1), x: stack! (b64) += y: stack! (b64);
        }
    }
    return x: stack! (b64)
}


total transformation time: REMOVED_TIME

total time (with save/print): REMOVED_TIME
